### ISSUE TITLE
SBT with 1.3.+ coursier fails to resolve dependency from custom uri based resolver 

### ISSUE THREAD

**Issue by YulawOne** [sbt/sbt#5390]
## steps
sbt version: 1.3.7 with coursier enabled

Our company are using custom GCS resolver implementation similar to https://github.com/lightbend/sbt-google-cloud-storage but it's closed source and I can't share many details

## problem
resolution fails with `checksum format error` on `*.pom__sha1` file during `*.pom` resolution

```
[error] Error downloading com.*:*_2.12:3.1.0-SNAPSHOT
[error]   Not found
[error]   Not found
[error]   not found: /Users/alexander/.ivy2/local/*/3.1.0-SNAPSHOT/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/*/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
[error]   checksum format error: /Users/alexander/.coursier/cache/v1/gcs/*/*_2.12/3.1.0-SNAPSHOT/.*_2.12-3.1.0-SNAPSHOT.pom__sha1
[error]   not found: gcs://repo.*.com/releases/*/*_2.12/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
[error]   not found: https://oss.sonatype.org/content/repositories/releases/*/*_2.12/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
[error]   not found: https://oss.sonatype.org/content/repositories/snapshots/*/*_2.12/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
```

## expectation
should work without errors

## notes
sbt 1.2.8 and sbt-coursier 1.1.0-M14 works without issues

**Comment by eed3si9n** [comment]
@YulawOne Thanks for the report. To debug, we need as much information as possible. Could you copy-paste the full error as it's printed by sbt please?

This sounds like something specific to Coursier, so please also report to coursier/coursier.

> `checksum format error`

If we read what the error message is saying, there's a formatting error with the checksum. Do you publish `pom.sha1` files? If so could you please copy-paste the content of them?

One workaround in case Coursier doesn't work at all is to fallback to Apache Ivy as follows:

```scala
ThisBuild / useCoursier := false
```

Would that work?

**Comment by YulawOne** [comment]
> If we read what the error message is saying, there's a formatting error with the checksum. Do you 
> publish pom.sha1 files? If so could you please copy-paste the content of them?

Yes, we publish in standard maven way, I can only share checksum as libraries are closed source `5524cd339c019ec88a089daeb429b28752e884f7`, the only thing might be is the file doesn't have line termination in the end.

> > ThisBuild / useCoursier := false
> 
> Would that work?

Yes, it works, but with ivy we're encountering issues with Intellij Idea and snapshots caching, for some reason idea are removing snapshot dependencies on re-import and cause long re-download each time

> @YulawOne Thanks for the report. To debug, we need as much information as possible. Could you copy-paste the full error as it's printed by sbt please?

I've tried to run with `set logLevel := Level.Debug` and `set every traceLevel := 0` but console doesn't show any internal to coursier exceptions, only:
```
[error] Error downloading com.*:*_2.12:3.1.0-SNAPSHOT
[error]   Not found
[error]   Not found
[error]   not found: /Users/alexander/.ivy2/local/*/3.1.0-SNAPSHOT/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/*/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
[error]   checksum format error: /Users/alexander/.coursier/cache/v1/gcs/*/*_2.12/3.1.0-SNAPSHOT/.*_2.12-3.1.0-SNAPSHOT.pom__sha1
[error]   not found: gcs://repo.*.com/releases/*/*_2.12/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
[error]   not found: https://oss.sonatype.org/content/repositories/releases/*/*_2.12/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
[error]   not found: https://oss.sonatype.org/content/repositories/snapshots/*/*_2.12/3.1.0-SNAPSHOT/*_2.12-3.1.0-SNAPSHOT.pom
```
I've ommited parts of uri related to our codebase.
When I went to check content of `.*_2.12-3.1.0-SNAPSHOT.pom__sha1` it was just empty file without content

**Comment by YulawOne** [comment]
It was a bug with our plugin, but was hard to identify due to silent behavior of coursier
